grakn-graph/src/main/java/ai/grakn/graph/internal/RelationTypeImpl.java

```diff
@@ -80,6 +76,13 @@ public Relation addRelation() {
     public RelationType hasRole(RoleType roleType) {
         checkTypeMutation();
         putEdge(roleType, Schema.EdgeLabel.HAS_ROLE);
+
+        //Cache the Role internally
+        cachedHasRoles.get().add(roleType);
```

Comment:
If you call `hasRole` while the roles are not cached, this will cause them to be looked up from the graph. Obviously not a big deal, but still an unnecessary lookup.